lib/src/span.dart

```diff
@@ -179,4 +180,56 @@ extension SourceSpanExtension on SourceSpan {
               primaryColor: primaryColor,
               secondaryColor: secondaryColor)
           .highlight();
+
+  /// Returns a span from [start] characters (inclusive) to [end] characters
```

Comment:
I'm a little worried about this API being confusing if there are different interpretations about what "characters" means. Here it looks like we are considering character to be a UTF 16 code points. In the `FileSpan` is it a byte offset instead?

I think this is an existing problem, but I'm not sure if adding this makes the problem worse.

Reply:
I believe everything in this package works in terms of code units, including `SourceFile` (with the exception of how `SourceFile.decoded()` behaves when passed numbers larger than `0xFFFF` but that's explicitly deprecated). This is the same—a character is a single UTF-16 code unit. I've updated the docs to be more explicit about that.